### PR TITLE
Fix ``bad-super-call`` for non-direct parents

### LINKED ISSUE #2903: False positive: calling super on bases base class is shown as error
### Steps to reproduce
1. Make an empty folder and create the file test.py with this content:
image
2. `pylint test.py`

### Current behavior
Executing the code works, but a false positive error is shown.

`E1003: Bad first argument 'B' given to super() (bad-super-call)`

### Expected behavior
Pylint shouldn't show this error.

### pylint --version output
```pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 26 2019, 21:43:19) 
[GCC 8.2.1 20181127]
```

### LINKED ISSUE #4922: E 1003: bad-super-call with two-argument super() call
### Bug description

```python
"""Produces 'bad-super-call'."""


class ClassB:
    "Dummy B."
    
    def whoami(self):
        """Print Name"""
        print("I am B")


class ClassC:
    "Dummy C."

    def whoami(self):
        """Print Name"""
        print("hello? this is C.")


class ClassD(ClassB, ClassC):
    "Dummy D."

    def greet(self):
        """Print Name"""
        super(ClassB, self).whoami()

d = ClassD()
d.greet()
d.whoami()
```

### Configuration

```ini
`super(ClassB, self).whoami()`
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
...
test.py:25:8: E1003: Bad first argument 'ClassB' given to super() (bad-super-call)
...
```

### Expected behavior

Hi!

This is similar to #2903, but the major difference is multiple inheritance. I hope the example makes sense from a code design perspective (if not just imagine that classB and classC inherit from an ABC that has the `whoami` method as an abstractmethod).

As far as I can tell, the flagged call to super is valid and the only way to actually call the `whoami` function of `ClassC` on `ClassD` instances.

Of course, in this example, switching the order of inheritance would be a simple solution, but just assume that based on other (not included) methods, the inheritance order has to be this way.

Thanks

### Pylint version

```shell
pylint 2.10.3-dev0
astroid 2.7.2
Python 3.9.6 (default, Jun 30 2021, 10:22:16) 
[GCC 11.1.0]
```

### OS / Environment

Linux 5.13.12-arch1-1 #1 SMP PREEMPT Wed, 18 Aug 2021 20:49:03 +0000 x86_64 GNU/Linux

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Add an entry to the change log describing the change in
  `doc/whatsnew/2/2.15/index.rst` (or ``doc/whatsnew/2/2.14/full.rst``
   if the change needs backporting in 2.14). If necessary you can write
   details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #4922, Closes #2903

### CODE REVIEW

File: doc/data/messages/b/bad-super-call/bad.py [hunk 1]
```diff
@@ -2,6 +2,11 @@ class Animal:
     pass
 
 
+class Fish:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-super-call/bad.py:5]:
```suggestion
class Fish(Animal):
```
Shouldn't Fish inherit from Animal ?

File: doc/data/messages/b/bad-super-call/good.py [hunk 2]
```diff
@@ -2,6 +2,10 @@ class Animal:
     pass
 
 
+class Fish:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-super-call/good.py:5]:
```suggestion
class Fish(Animal):
```

File: tests/functional/s/super/super_checks.py [hunk 3]
```diff
@@ -123,3 +123,34 @@ class SuperWithSelfClass(object):
...
+class GrandChild(Child):
+    def method(self):
+        print("Grandchild")
+        super(GrandChild, self).method()
+        super(Child, self).method()
+        super(Niece, self).method()  # [bad-super-call]
+
+
+# Reported in https://github.com/PyCQA/pylint/issues/4922
+class AlabamaCousin(Child, Niece):
```
Comment (Pierre-Sassoulas) [inline tests/functional/s/super/super_checks.py:153]: Well, that's very funny 😄 ``CarlosTwoOfSpain`` works too if we want to keep the goodwill of alabamians 😛

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/data/messages/b/bad-super-call/bad.py (hunk 1 above)
Comment (DanielNoord) [reply doc/data/messages/b/bad-super-call/bad.py:5]: Yeah, but then this example doesn't work.

I'll change!

```suggestion
class Tree:
```

File: doc/data/messages/b/bad-super-call/good.py (hunk 2 above)
Comment (DanielNoord) [reply doc/data/messages/b/bad-super-call/good.py:5]:
```suggestion
class Tree:
```

File: doc/data/messages/b/bad-super-call/bad.py [hunk 4]
```diff
@@ -2,6 +2,11 @@ class Animal:
...
 
 
+class Tree:
+    pass
+
+
 class Cat(Animal):
     def __init__(self):
-        super(Animal, self).__init__()  # [bad-super-call]
+        super(Fish, self).__init__()  # [bad-super-call]
```
Comment (DanielNoord) [inline doc/data/messages/b/bad-super-call/bad.py:11]:
```suggestion
        super(Tree, self).__init__()  # [bad-super-call]
```

File: doc/data/messages/b/bad-super-call/bad.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/data/messages/b/bad-super-call/bad.py]: I did suggest tree at first too but the Niece/parent example made me think this should work with fish/animal ?

Comment (DanielNoord) [reply doc/data/messages/b/bad-super-call/bad.py]: Don't you think this is clearer?

Comment (DanielNoord) [reply doc/data/messages/b/bad-super-call/bad.py]: Let's wait for the implementation of #6961 before adding that.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**